src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1202,6 +1209,95 @@ DatabaseShardImp::findAcquireIndex(
     return boost::none;
 }
 
+void
+DatabaseShardImp::scheduleFinalizeShard(std::uint32_t shardIndex)
+{
+    taskQueue_->addTask([this, shardIndex]
+    {
+        auto retry = [this, shardIndex]
+        {
+            const uint8_t TIMER_INTERVAL = 60;
+
+            using timer = boost::asio::basic_waitable_timer<
+                std::chrono::steady_clock>;
+
+            std::shared_ptr<timer> retryTimer(new timer(app_.getIOService()));
+
+            retryTimer->expires_from_now(std::chrono::seconds(TIMER_INTERVAL));
+            retryTimer->async_wait([this, shardIndex, retryTimer]
```

Comment:
Should be able to move `retryTimer` into the lambda eg. `retryTimer{std::move(retryTimer)}`

Reply:
The timers need to be canceled in `DatabaseShardImp::onStop()`

Reply:
Done with the unique_ptr reset in your patch